### ISSUE TITLE
Make user names cleaner

### ISSUE THREAD

**Issue by adelavega** [neurostuff/neurostore#357]
Currently they are ugly oauth2 IDs

**Comment by nicoalee** [comment]
- [ ] change user names to name instead of user.sub

**Comment by nicoalee** [comment]
@jdkent I don't think this task is super straighforward. When a user clone a study, we save their given auth0 ID `(user.sub)`. 

The easiest solution would be to save the user's name/username when they clone the study. However, this may eventually be out of date if the user changes their name/username at a later time.

**Comment by jdkent** [comment]
I think Auth0 has a mechanism to access the name field.
Here is an example for the development server:

## get name field from Auth0 (using the management api token)
https://dev-mui7zm42.us.auth0.com/api/v2/users/auth0%7C62e0e6c9dd47048572613b4d?fields=name&include_fields=true

## Management API token
using the auth0 management api token from here: https://manage.auth0.com/dashboard/us/dev-mui7zm42/apis/60f60987000aa700428537c5/test

What could be a solution is adding an endpooint to /users/ where a `PUT` request would call auth0 to update there and update our database at the same time.

**Comment by nicoalee** [comment]
@jdkent It's trivial to get the name of the user currently logged in; however, what about the list of studies that have a sourceId (a user cloned the study)?
When we display public studies in the search page, those other auth0 IDs belonging to other users will be displayed.

![image](https://user-images.githubusercontent.com/26612023/191058778-1aefa308-9fb4-40f6-b202-ced2b9c6a895.png)

**Comment by nicoalee** [comment]
> What could be a solution is adding an endpooint to /users/ where a `PUT` request would call auth0 to update there and update our database at the same time.

Another solution would just be to do a mapping on the frontend. We can gather all userIDs that are displayed on the page and call the management API: https://auth0.com/docs/manage-users/user-search/retrieve-users-with-get-users-endpoint

**Comment by nicoalee** [comment]
@jdkent @nicoalee 

Update:

Our solution will be to create a `displayName `column in the backend user table.
This property will be added to the study return object and any other object that the user owns (studysets, meta-analyses, etc)

This will replace the FE auth0IDs for a more user friendly visual.

In the future, we will want to be able to update their profile and give them the ability to update their display name

**Comment by nicoalee** [comment]
still a WIP

**Comment by adelavega** [comment]
Related: #487

**Comment by nicoalee** [comment]
completed via https://github.com/neurostuff/neurostore/issues/197
